src/transformer-components/SavedTransformerView.tsx

```diff
@@ -15,19 +29,121 @@ function SavedTransformerView({
   transformer: SavedTransformer;
 }): ReactElement {
   const [errMsg, setErrMsg] = useState<string | null>(null);
+  const [editable, setEditable] = useState<boolean>(false);
+  const [savedTransformer, setSavedTransformer] = useState(urlTransformer);
+  const [saveErr, setSaveErr] = useState<string | null>(null);
+
+  // Load saved state from CODAP memory
+  useEffect(() => {
+    async function fetchSavedState() {
+      const savedState = (await getInteractiveFrame()).savedState;
+      if (savedState && savedState.savedTransformation) {
+        setSavedTransformer({
+          ...savedTransformer,
+          name: savedState.savedTransformation.name,
+          description: savedState.savedTransformation.description,
+        });
+      }
+    }
+    fetchSavedState();
+  }, []);
+
+  // Register a listener to generate the plugin's state
+  useEffect(() => {
+    const callback = (
+      previousInteractiveState: InteractiveState
+    ): InteractiveState => {
+      return {
+        ...previousInteractiveState,
+        savedTransformation: {
+          name: savedTransformer.name,
+          description: savedTransformer.description || "",
+        },
+      };
+    };
+
+    addInteractiveStateRequestListener(callback);
+    return () => removeInteractiveStateRequestListener(callback);
+  }, [savedTransformer]);
+
+  function notifyStateIsDirty() {
+    notifyInteractiveFrameIsDirty();
+  }
 
   return (
     <div className="transformer-view">
-      <h2>
-        {urlTransformer.name}
-        <span id="transformerBase"> ({urlTransformer.content.base})</span>
-      </h2>
-      {urlTransformer.description && <p>{urlTransformer.description}</p>}
+      {editable && (
+        <div className="editing-indicator">
+          <p>You are editing this transformer...</p>
+        </div>
+      )}
+      {editable ? (
+        <TextInput
+          value={savedTransformer.name}
+          onChange={(e) => {
+            setSavedTransformer({ ...savedTransformer, name: e.target.value });
+            setSaveErr(null);
+          }}
+          placeholder={"Transformer Name"}
+          className="saved-transformer-name"
+          onBlur={notifyStateIsDirty}
+        />
+      ) : (
+        <h2>
+          {savedTransformer.name}
+          <span id="transformerBase"> ({savedTransformer.content.base})</span>
+        </h2>
+      )}
+      {editable ? (
+        <TextArea
+          value={savedTransformer.description || ""}
+          onChange={(e) => {
+            setSavedTransformer({
+              ...savedTransformer,
+              description: e.target.value,
+            });
+            setSaveErr(null);
+          }}
+          placeholder="Purpose Statement"
+          className="purpose-statement"
+          onBlur={notifyStateIsDirty}
+        />
+      ) : (
+        savedTransformer.description && <p>{savedTransformer.description}</p>
+      )}
```

Comment:
Can we add an `<hr/>` between the name/purpose statement and the actual transformer data so when you're editing a saved transformation it's clear which boxes are the transformation and which are name/purpose statement.

Reply:
Always or just while you're editing?

Reply:
My instinct is just while editing as the styling we use when not editing is sufficient to distinguish the two sections, but if it looks better to have it always then that works to.